File: BACKEND/user-service/src/main/java/kj001/user_service/dtos/FaceDataDTO.java
package kj001.user_service.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.web.multipart.MultipartFile;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class FaceDataDTO {
    private Long userId;
    private String faceEncoding;
    private MultipartFile faceImage;  // Used for request only
    private String originalImagePath;
    private boolean hasFaceData;
}

File: FLUTTER/lib/features/user/data/models/FaceDataDTO.dart
class FaceDataDTO {
  final int userId;
  final String faceEncoding;
  final String originalImagePath;
  final bool hasFaceData;

  FaceDataDTO({
    required this.userId,
    required this.faceEncoding,
    required this.originalImagePath,
    required this.hasFaceData,
  });

  factory FaceDataDTO.fromJson(Map<String, dynamic> json) {
    return FaceDataDTO(
      userId: json['userId'] ?? 0,
      faceEncoding: json['faceEncoding'] ?? "",
      originalImagePath: json['originalImagePath'] ?? "",
      hasFaceData: json['hasFaceData'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'faceEncoding': faceEncoding,
      'originalImagePath': originalImagePath,
      'hasFaceData': hasFaceData,
    };
  }
}